### ISSUE TITLE
[Exynos S9/Note9/S10 Series] Transitions/Animations broken 

### ISSUE THREAD

**Issue by malinathani** [phhusson/treble_experimentations#595]
Basically UI animations are white/washed out and its the same issue on both Exynos 9810 and 9820 devices. 
[S10_hwc_logcat (1).txt](https://github.com/phhusson/treble_experimentations/files/3355279/S10_hwc_logcat.1.txt)
[SurfaceFlinger_S10+_help_thx (1).txt](https://github.com/phhusson/treble_experimentations/files/3355280/SurfaceFlinger_S10%2B_help_thx.1.txt)

**Comment by phhusson** [comment]
Please dump `dumpsys SurfaceFlinger` during animation and after

Le mer. 3 juil. 2019 à 15:53, malinathani <notifications@github.com> a
écrit :

**Comment by malinathani** [comment]
During animation taken during swipe up, hope this helps

[during_animation.txt](https://github.com/phhusson/treble_experimentations/files/3356196/during_animation.txt)
[post_animation.txt](https://github.com/phhusson/treble_experimentations/files/3356197/post_animation.txt)

**Comment by penn5** [comment]
Try with animators set to 10x in settings. It should make it *much* easier to debug, as all animations take 10x as long.

**Comment by ktshikotshi** [comment]
Disabling HW overlays on developer settings fixes the issue, but you'll need to do it on every boot

**Comment by malinathani** [comment]
Yes that fixes the issue but that's not the solution, it just makes the phone very laggy and unusable.

**Comment by phhusson** [comment]
Only thing I can see for the moment is that during animation, there is a new section that appears "mHasCompositionLayer(1)", which is at 0/empty out-of-animation, and it has internal_format 0x1, while in client-mode it has internal_format 0x100000001.

It would be nice if someone could take the `dumpsys SurfaceFlinger` during the same animations.

I'd be curious to force in DCI-P3 mode to see what happens

**Comment by phhusson** [comment]
Using split-screen (in my case settings + chrome) triggers the bug without animation.

Interestingly if pressing finger on the separator, the screens goes right

**Comment by malinathani** [comment]
yeah the bug is not just limited to that animation

**Comment by phhusson** [comment]
Setting persist.sys.sf.native_mode from `1` (MANAGED, default value from vendor) to `0` (UNMANAGED) gives the white effect on all surfaces. (can be set on the fly with `service call SurfaceFlinger 1023 i32 0`)

"CompositionInfo/mHasCompositionLayer" (which seem to refer to secondary M2M blitter, which is seen when the bug is triggered) seem identical between stock ROM and GSI (dataSpace, internal_format, assignedMPP, M2M, ... are identical)

**Comment by phhusson** [comment]
This hack works:
https://gist.github.com/phhusson/79d9edf69f62e3c1ab1cca3c5159d396

this HAL_DATASPACE_SRGB_LINEAR with HAL_DATASPACE_UNKNOWN

According to the HAL it shouldn't change anything:
https://android.googlesource.com/platform/hardware/interfaces/+/refs/heads/master/graphics/common/1.0/types.hal

    SRGB_LINEAR = STANDARD_BT709 | TRANSFER_LINEAR | RANGE_FULL,
    DATASPACE_UNKNOWN = STANDARD_UNSPECIFIED | TRANSFER_UNKNOWN | RANGE_UNKNOWN

STANDARD_UNSPECIFIED it said to be bt709 in RGB
TRANSFER_UNKNOWN is TRANSFER_SRGB
RANGE_UNKNOWN is RANGE_FULL

Forcing HAL_DATASPACE_SRGB instead of HAL_DATASPACE_SRGB_LINEAR also fixes the issue.

My guess is that Samsung's framework has been built with `ANDROID_ENABLE_LINEAR_BLENDING`

**Comment by Fzndox** [comment]
This issue appears to be fixed under v216.
